server/src/instant/util/storage.clj

```diff
@@ -0,0 +1,207 @@
+                           "data" (cel/->cel-map {"path" filepath})})))))
+
+
+(defn upload-image-to-s3 [app-id filename image-url]
+  (let [object-key (->object-key app-id filename)]
+    (storage-beta/assert-storage-enabled! app-id)
+    (s3-util/upload-image-to-s3 object-key image-url)))
+
+(defn create-signed-download-url!
+  ([{:keys [app-id filename refresh-token rules-override]}]
+   (assert-storage-permission! "view" {:app-id app-id
+                                       :refresh-token refresh-token
+                                       :filepath filename
+                                       :rules-override rules-override})
+   (create-signed-download-url! app-id filename))
+  ([app-id filename refresh-token]
+   (assert-storage-permission! "view" {:app-id app-id
+                                       :refresh-token refresh-token
+                                       :filepath filename})
+   (create-signed-download-url! app-id filename))
+  ([app-id filename]
+   (let [expiration (+ (System/currentTimeMillis) (* 1000 60 60 24 7)) ;; 7 days
+         object-key (->object-key app-id filename)]
+
+     (storage-beta/assert-storage-enabled! app-id)
+     (str (s3-util/signed-download-url object-key expiration)))))
+
+(defn create-signed-upload-url!
+  ([{:keys [app-id filename refresh-token rules-override]}]
+   (assert-storage-permission! "create" {:app-id app-id
+                                         :refresh-token refresh-token
+                                         :filepath filename
+                                         :rules-override rules-override})
+   (create-signed-upload-url! app-id filename))
+  ([app-id filename refresh-token]
+   (assert-storage-permission! "create" {:app-id app-id
+                                         :refresh-token refresh-token
+                                         :filepath filename})
+   (create-signed-upload-url! app-id filename))
+  ([app-id filename]
+   (let [object-key (->object-key app-id filename)]
+     (storage-beta/assert-storage-enabled! app-id)
+     (str (s3-util/signed-upload-url object-key)))))
+
+(comment
+  (def app-id  #uuid "524bc106-1f0d-44a0-b222-923505264c47")
+  (def filename "demo.png")
+  (def image-url "https://i.redd.it/bugxrdkjmm1b1.png")
+  (create-signed-upload-url! app-id filename)
+  (upload-image-to-s3 app-id filename image-url)
+  (create-signed-download-url! app-id filename))
+
+(defn format-object [{:keys [key size owner etag last-modified]}]
+  {:key key
+   :name (object-key->filename key)
+   :size size
+   :owner owner
+   :etag etag
+   :last_modified (.getMillis last-modified)})
+
+;; Retrieves all files that have been uploaded via Storage APIs
+(defn list-files! [app-id subdirectory]
+  (let [_ (storage-beta/assert-storage-enabled! app-id)
+        prefix (if (string/blank? subdirectory)
+                 app-id
+                 (str app-id "/" subdirectory))
+        objects-resp (s3-util/list-app-objects prefix)
```

Comment:
This has an implicit limit of 1000, right?

Reply:
yeahhh planning on handling pagination in a followup!